### PR TITLE
Update CMake configuration to reflect latest changes

### PR DESCRIPTION
**Pull request by marbre**
Seems that #226 was closed accidentally.

### CODE REVIEW

**Comment by marbre** [comment]
@GMNGeoffrey May I request a review from you, if you're not on vacation?

**Comment by GMNGeoffrey** [comment]
> Seems that #226 was closed accidentally.

Oh you're right! Our automation is fighting with MLIR's. It was tagged to close their #226, not ours :-/ Ugh sorry this is so janky.

I can take a look, but I'm not super familiar with cmake. Maybe @benvanik would be a better reviewer?

**Comment by marbre** [comment]
> Oh you're right! Our automation is fighting with MLIR's. It was tagged to close their #226, not ours :-/ Ugh sorry this is so janky.

Fortunately, I think this won't be a problem in future as MLIR has landed in LLVM.

> I can take a look, but I'm not super familiar with cmake. Maybe @benvanik would be a better reviewer?

Being back in office after New Year's Eve, I hope to get the tests working with CMake within the first week of January. Therefore, it would make work a little bit easier if some of my PRs can make it into the master before. But even if not, I can handle it as long as the changes gets merged someday ;)

File: iree/base/CMakeLists.txt [hunk 1]
```diff
@@ -137,43 +152,69 @@ iree_cc_library(
   NAME
     file_io
   SRCS
+    #(marbre): Pulled in as dep defined by platform_trampoline_deps("file_io") in the Bazel world
```
Comment (GMNGeoffrey) [inline iree/base/CMakeLists.txt]: Is this a TODO for future work? I haven't seen the "#(username)" syntax before. If it's just a comment, maybe leave off the username?

Comment (marbre) [reply iree/base/CMakeLists.txt]: Yeah, you're right. Missed the TODO, but this will be superseded by #227 (with #239 as a first step) quite soon..

Comment (GMNGeoffrey) [reply iree/base/CMakeLists.txt]: Let's add the TODOs then. If you're mobile or something, I'm happy to add a commit on top of this to do it (and I'll fix the authorship when it gets imported :smile:)

**Comment by GMNGeoffrey** [comment]
FYI, the CMake build now runs on presubmit. I think if you rebase all these commits onto master it will run (for those that touch parts that are enabled)

**Comment by marbre** [comment]
> FYI, the CMake build now runs on presubmit. I think if you rebase all these commits onto master it will run (for those that touch parts that are enabled)

Yeah, thanks for the info. I was already pleased to noticed :D

File: iree/base/CMakeLists.txt (hunk 1 above)
Comment (marbre) [reply iree/base/CMakeLists.txt]: Fixed it myself :wink:

**Pull request closed without merging**